src/vdaf/poplar1.rs

```diff
@@ -308,6 +381,44 @@ enum SketchState<F> {
     RoundTwo,
 }
 
+impl<F: ConstantTimeEq> PartialEq for SketchState<F> {
+    fn eq(&self, other: &Self) -> bool {
+        self.ct_eq(other).into()
+    }
+}
+
+impl<F: ConstantTimeEq> Eq for SketchState<F> {}
+
+impl<F: ConstantTimeEq> ConstantTimeEq for SketchState<F> {
+    fn ct_eq(&self, other: &Self) -> Choice {
+        // We allow short-circuiting on the round (RoundOne vs RoundTwo).
+        match (self, other) {
+            (
+                SketchState::RoundOne {
+                    A_share: self_a_share,
+                    B_share: self_b_share,
+                    is_leader: self_is_leader,
+                },
+                SketchState::RoundOne {
+                    A_share: other_a_share,
+                    B_share: other_b_share,
+                    is_leader: other_is_leader,
+                },
+            ) => {
+                let self_is_leader = Choice::from(*self_is_leader as u8);
+                let other_is_leader = Choice::from(*other_is_leader as u8);
```

Comment:
Short-circuiting on the is_leader should be fine, too. In fact, I would do this just to make clear to the reader what is secret and what is not.